plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -107,21 +105,20 @@ private void testAnalyze(Optional<Integer> checkpointInterval)
                         "('name', 354.0, 25.0, 0.0, null, null, null)," +
                         "(null, null, null, null, 50.0, null, null)");
 
-        // insert modified rows
-        assertUpdate("INSERT INTO " + tableName + " SELECT nationkey + 25, reverse(name), regionkey + 5, reverse(comment) FROM tpch.sf1.nation", 25);
+        // check that analyze does not change already calculated statistics
+        assertUpdate("ANALYZE " + tableName);
 
-        // without ANALYZE all stats but size and NDV should be updated
         assertQuery(
                 "SHOW STATS FOR " + tableName,
                 "VALUES " +
-                        "('nationkey', null, 25.0, 0.0, null, 0, 49)," +
-                        "('regionkey', null, 5.0, 0.0, null, 0, 9)," +
+                        "('nationkey', null, 25.0, 0.0, null, 0, 24)," +
+                        "('regionkey', null, 5.0, 0.0, null, 0, 4)," +
                         "('comment', 3714.0, 25.0, 0.0, null, null, null)," +
                         "('name', 354.0, 25.0, 0.0, null, null, null)," +
-                        "(null, null, null, null, 75.0, null, null)");
+                        "(null, null, null, null, 50.0, null, null)");
 
-        // with analyze we should get new size and NDV
```

Comment:
do we have coverage for ANALYZE filling in the NDV information if it was not present before?
should we run the above INSERT with stats collection disabled?

Reply:
I think NDV information update by ANALYZE is covered in testStatisticsOnInsertWhenCollectionOnWriteDisabled.
Filling previously empty NVD with ANALYZE is covered in testCreateTableStatisticsWhenCollectionOnWriteDisabled.